test/integration/apiserver/openapi/openapiv3_test.go

```diff
@@ -214,6 +214,15 @@ func TestOpenAPIV3ProtoRoundtrip(t *testing.T) {
 		t.Fatal(err)
 	}
 
+	// OpenAPI v3 proto definitions have only three types of defaults: number, boolean, string. It doesn't have struct defaults.
+	// In the JSON/YAML OpenAPI definitions, however, there are cases with "default": {}, deserializing into JSON as map[string]any.
+	// From protobuf, however, these deserialize naturally into nil, as they are not expressed.
+	// We should probably patch the OpenAPI v3 spec to not include any empty struct defaults at all in JSON form, as proto doesn't support it.
+	// For now, however, we can ignore these differences.
```

Comment:
Actually, every JSON value could show up as default. How do we ensure these never hit these (outside of tests) ?

Reply:
You mean that someone would put `"default": {"foo": "bar"}` into the OpenAPI?

The other alternative is changing proto to support struct defaults I guess. Does that sound feasible?

Reply:
Tested this now with CRDs as well. The findings:
a) native types have at least "simple" defaults that proto support, e.g. string, number, bool. These show up in OpenAPI
b) if one specified defaults in a CRD OpenAPI schema declaratively, these show up in server's OpenAPI schema as well as you'd expect
c) _however_ CRD declarative defaulting can be non-simple types, sth like `{"cronSpec": "5 0 * * *", "image": "foo"}`. This shows up correctly in JSON OpenAPI v3, but NOT at all in proto, here's the diff from this test in that case:

```
json is -
proto is +
         						"spec": {
          							VendorExtensible: {},
          							SchemaProps: spec.SchemaProps{
          								... // 6 identical fields
          								Format:           "",
          								Title:            "",
        - 								Default:          map[string]any{"cronSpec": string("5 0 * * *"), "image": string("foo")},
        + 								Default:          nil,
          								Maximum:          nil,
          								ExclusiveMaximum: false,
          								... // 24 identical fields
          							},
          							SwaggerSchemaProps: {},
          							ExtraProps:         nil,
```

Reply:
Have double checks that we expose defaults of CRDs. So how do these show up in a proto serialization?